src/datumaro/plugins/sam_transforms/interpreters/sam_decoder_for_amg.py

```diff
@@ -0,0 +1,219 @@
+        masks: Array of masks corresponded to the points.
+        iou_preds: Array of Intersection over Union (IoU)
+            prediction scores corresponded to the points.
+    """
+
+    masks: np.ndarray = field()
+    iou_preds: np.ndarray = field()
+
+    @classmethod
+    def cat(cls, masks: List["AMGMasks"]) -> "AMGMasks":
+        """Concatenate a list of `AMGMasks` into a single `AMGMasks` object.
+
+        Parameters:
+            masks: List of `AMGMasks` to concatenate.
+
+        Returns:
+            A new AMGMasks containing the concatenated masks and IoU prediction scores.
+        """
+        return AMGMasks(
+            masks=np.concatenate([mask.masks for mask in masks], axis=0),
+            iou_preds=np.concatenate([mask.iou_preds for mask in masks], axis=0),
+        )
+
+    def postprocess(
+        self,
+        mask_threshold: float,
+        pred_iou_thresh: float,
+        stability_score_offset: float,
+        stability_score_thresh: float,
+        box_nms_thresh: float,
+        min_mask_region_area: int,
+    ) -> List[Mask]:
+        """Postprocesses the masks with the given parameters.
+
+        Parameters:
+            pred_iou_thresh (float): A filtering threshold in [0,1], using the
+                model's predicted mask quality.
+            stability_score_thresh (float): A filtering threshold in [0,1], using
+                the stability of the mask under changes to the cutoff used to binarize
+                the model's mask predictions.
+            stability_score_offset (float): The amount to shift the cutoff when
+                calculated the stability score.
+            box_nms_thresh (float): The box IoU cutoff used by non-maximal
+                suppression to filter duplicate masks.
+            min_mask_region_area (int): If >0, postprocessing will be applied
+                to remove the binary mask which has the number of 1s less than min_mask_region_area.
+
+        Returns:
+            List of :class:`Mask`s representing the postprocessed masks.
+        """
+        masks, iou_preds = self.masks, self.iou_preds
+
+        if pred_iou_thresh > 0.0:
+            keep_mask = iou_preds > pred_iou_thresh
+            masks = masks[keep_mask]
+            iou_preds = iou_preds[keep_mask]
+
+        if stability_score_thresh > 0.0:
+            keep_mask = (
+                self._calculate_stability_score(
+                    masks=masks,
+                    mask_threshold=mask_threshold,
+                    stability_score_offset=stability_score_offset,
+                )
+                > stability_score_thresh
+            )
+            masks = masks[keep_mask]
+            iou_preds = iou_preds[keep_mask]
+
+        binary_masks = masks > mask_threshold
+
+        if min_mask_region_area > 0:
+            keep_mask = binary_masks.sum(axis=(1, 2)) > min_mask_region_area
+            binary_masks = binary_masks[keep_mask]
+            iou_preds = iou_preds[keep_mask]
```

Comment:
These threshold (`pred_iou_threshold`, `stability_score_threshold`, and `min_mask_region_area`) looks like given by a user-end. How about enforcing them to be positive when creating `SAMAutomaticMaskGeneration` instead of these if conditions.

Reply:
Can I patch it in the following PR? This is because I'm hardly revisiting Arrow format for now so that checkout this branch and doing some works make me confusing.

Reply:
Sure, please.